ui/app/routes/vault/cluster/auth.js

```diff
@@ -101,19 +126,24 @@ export default class AuthRoute extends ClusterRouteBase {
     In older versions of Vault, the "with" query param could refer to either the auth mount path or the type
     (which may be the same, since the default mount path *is* the type). 
     For backward compatibility, we handle both scenarios.
-    → If `authMount` matches a visible auth mount, return its mount data (which includes the type).
-    → If it matches a supported auth type instead, return just the type to preselect it in the dropdown.
+    → If `authMount` matches a visible auth mount the method will assume that mount path to login and render as the default in the login form.
+    → If `authMount` matches a supported auth type (and the mount does not have `listing_visibility="unauth"`), that is preselected in the login form.
   */
   getMountOrTypeData(authMount, visibleAuthMounts) {
-    if (visibleAuthMounts?.[authMount]) {
-      return { path: authMount, ...visibleAuthMounts[authMount], isVisibleMount: true };
+    const sanitizedParam = sanitizePath(authMount); // strip leading/trailing slashes
```

Comment:
added test [here](https://github.com/hashicorp/vault/pull/30700/files?diff=split&w=1#diff-7c072458c21367a973345b1a8a712fa34739b8a59657dc78cac51ed55e552e3fR58-R66) for this update